### DISCUSSION TITLE
Set-PnPTraceLog does not log provisioning debug info in Azure Function

### DISCUSSION THREAD

**Discussion by UMAknow-Patrick** [pnp/powershell#476] [Q&A]
Hi everyone,

I've tested the provisioning of an online site with code I tested locally in PowerShell 7.1
But no luck within an Azure Function (~3) the log file is always empty:
```powershell
# Input bindings are passed in via param block.
param([string] $QueueItem, $TriggerMetadata)

# Write out the queue message and insertion time to the information log.
Write-Information "PowerShell queue trigger function processed work item: $QueueItem"
Write-Information "Queue item insertion time: $($TriggerMetadata.InsertionTime)"

$encryptedPassword = ConvertTo-SecureString $env:USER_PASSWORD -AsPlainText -Force
$userCredentials = New-Object System.Management.Automation.PsCredential($env:USER_ACCOUNT, $encryptedPassword)

$logFilePath = "$PSScriptRoot\logs\provisioningLog.txt"
$template1 = "$PSScriptRoot\templates\common-uploadFile.xml"
$template2 = "$PSScriptRoot\templates\common-sitecol-contenttype.xml"
Set-PnPTraceLog -On -LogFile $logFilePath -Level Debug

Connect-PnPOnline -Url $env:SITE_URL -Credentials $userCredentials

Write-Information "Template: $template1"
Invoke-PnPSiteTemplate -Path $template1

Write-Information "Template: $template2"
Invoke-PnPSiteTemplate -Path $template2

Disconnect-PnPOnline
```
Same code (except `Invoke-PnPSiteTemplate` replaced by `Apply-PnPProvisioningTemplate`) for Azure function ~1 works perfectly.
Has anybody experienced empty trace log file problems with Azure function App version ~3 ?
Thanks

**Comment by jenunn** [comment]
Set-PnPTracelog does not work for me either in Azure Runbook or in Powershell with PnP.PowerShell v1.5....

**Comment by ocimran** [comment]
Doesn't work for me either

**Comment by orty** [comment]
Same here :/

**Comment by sadomovalex** [comment]
same here - it doesn't work in regular PowerShell as well. Tested with PnP.PowerShell 1.7.0:
```powershell
Set-PnPTraceLog -On -LogFile "output.log" -Level Debug
```
output log is empty. Since this is discussion which is probably not actively monitored by dev team I've created issue about it: https://github.com/pnp/powershell/issues/1061.

**Comment by Oegl** [comment]
I still have the problem that the trace log is empty - is there a workarround to get detailed information during provisioning?

**Reply by stokuri** [reply]
Set-PnPTraceLog -Off should close and flush the log contents. Once this is done, you should see something written to the trace file (depending on the earlier command executions - some of those invocations should produce some traces)

**Comment by stokuri** [comment]
I have the same problem. Potential fix(working for me) -> https://github.com/pnp/powershell/pull/2652/commits/df7745955976d47f7da992f14e9d98618f01fd92

Raised PR for the same.
